src/main/java/com/dtforce/spring/kubernetes/KubernetesDiscoveryClient.java

```diff
@@ -49,33 +44,80 @@ public String description()
 			return Collections.emptyList();
 		}
 
-		if (log.isDebugEnabled()) {
-			log.debug("getInstances: service = {}", service.toString());
+		return getInstancesFromService(service);
+	}
+
+	@Override
+	public List<ServiceInstance> selectInstances(Map<String, String> match)
+	{
+		return selectInstances(match, Collections.emptyMap());
+	}
+
+	@Override
+	public List<ServiceInstance> selectInstances(Map<String, String> match, Map<String, String> doNotMatch)
+	{
+		ServiceList serviceList;
+		try {
+			serviceList = kubeClient.services().withLabels(match).withoutLabels(doNotMatch).list();
+		} catch(KubernetesClientException e) {
+			log.warn("getInstancesMatchingMetadata: failed to retrieve matching services: API call failed: {}",
+				e.getMessage());
+			return Collections.emptyList();
 		}
 
-		if (service.getSpec().getPorts().isEmpty()) {
-			log.error("getInstances: service '{}' has no ports", serviceId);
+		if (serviceList == null) {
+			log.warn("getInstancesMatchingMetadata: service list is null");
```

Comment:
This on the other hand seems like perfectly normal case and logging warn seems unnecessary.

Reply:
A typical empty query response from the Kubernetes API (and in turn, the client) is an empty list, not a null value. I'll make this clear by adding a comment.